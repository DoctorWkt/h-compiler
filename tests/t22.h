#define EOF 4
int islegal(int c)
{
    if (c >= ' ') {
        if (c <= '~') {
            return(1);
        }
    }
    if (c == ' ') return(1);
    if (c == '\t') return(1);
    if (c == '\n') return(1);
    if (c == EOF) return(1);
    return(0);
}

void main()
{
  char ch;

  for (ch=-30; ch != 300; ch++) {
    if (islegal(ch)) {
      putchar(ch); putchar(' '); printn(ch);
    }
  }
  exit();
}
