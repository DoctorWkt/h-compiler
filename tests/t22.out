 000004
	 000011

 000012
  000040
! 000041
" 000042
# 000043
$ 000044
% 000045
& 000046
' 000047
( 000050
) 000051
* 000052
+ 000053
, 000054
- 000055
. 000056
/ 000057
0 000060
1 000061
2 000062
3 000063
4 000064
5 000065
6 000066
7 000067
8 000070
9 000071
: 000072
; 000073
< 000074
= 000075
> 000076
? 000077
@ 000100
A 000101
B 000102
C 000103
D 000104
E 000105
F 000106
G 000107
H 000110
I 000111
J 000112
K 000113
L 000114
M 000115
N 000116
O 000117
P 000120
Q 000121
R 000122
S 000123
T 000124
U 000125
V 000126
W 000127
X 000130
Y 000131
Z 000132
[ 000133
\ 000134
] 000135
^ 000136
_ 000137
` 000140
a 000141
b 000142
c 000143
d 000144
e 000145
f 000146
g 000147
h 000150
i 000151
j 000152
k 000153
l 000154
m 000155
n 000156
o 000157
p 000160
q 000161
r 000162
s 000163
t 000164
u 000165
v 000166
w 000167
x 000170
y 000171
z 000172
{ 000173
| 000174
} 000175
~ 000176
